cmd/mdatagen/templates/resource_test.go.tmpl

```diff
@@ -0,0 +1,63 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package {{ .Package }}
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+{{- $enabledAttrCount := 0 }}
+{{- range $name, $info := .ResourceAttributes }}
+{{- if $info.Enabled }}
+{{- $enabledAttrCount = inc $enabledAttrCount }}
+{{- end }}
+{{- end }}
+
+func TestResourceBuilder(t *testing.T) {
+	for _, test := range []string{"default", "all_set", "none_set"} {
+		t.Run(test, func(t *testing.T) {
+			cfg := loadResourceAttributesConfig(t, test)
+			rb := NewResourceBuilder(cfg)
+			{{- range $name, $info := .ResourceAttributes }}
+			rb.Set{{ $name.Render }}({{ $info.Type.TestValue }})
```

Comment:
nit: What about doing this (or something similar) so that we set different attributes to different values?
```suggestion
			rb.Set{{ $name.Render }}("{{ $name.Render }}")
```

Reply:
Good point, but the suggestion will be applied to string attributes only and will be inconsistent with metrics values. I think we need to tackle this separately and update the TestValue function instead. I submitted https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24370 and can take do it before merging this PR if you want to

Reply:
Submitted https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24373